sdk/servicebus/azure-servicebus/api_samples.py

```diff
@@ -0,0 +1,254 @@
+
+        # 1.2 send batch
+        batch_message = topic_sender.create_batch()
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        topic_sender.send(batch_message, session_id="test_session")
+
+        # 1.3 schedule
+        schedule_message = Message("Scheduled message")
+        schedule_time_utc = datetime.utcnow() + timedelta(minutes=10)
+        sequence_number = topic_sender.schedule(schedule_message, schedule_time_utc, session_id="test_session")
+
+        batch_schedule_message = topic_sender.create_batch()
+
+        batch_sequence_numbers = topic_sender.schedule(
+            message=batch_schedule_message,
+            schedule_time_utc=schedule_time_utc,
+            session_id="test_session"
+        )
+
+        # 1.4 cancel schedule
+        topic_sender.cancel_scheduled_messages(sequence_number)
+        topic_sender.cancel_scheduled_messages(batch_sequence_numbers)
+
+# 2 Receive
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+subscription_receiver = sb_client.get_subscription_receiver(
+    topic_name="topic_name",
+    subscription_name="subscription_name",
+    session_id="test_session"
+)
+
+# 2.1 iterator receive
+with sb_client:
+    with subscription_receiver:
+        session = subscription_receiver.session
+        session.set_session_state("START")
+        for msg in subscription_receiver:
+            if session.expired:
+                break
+            print(message)
+            msg.complete()
+            session.renew_lock()
+
+# 2.2 pull mode receive
+with sb_client:
+    with subscription_receiver:
+        session = subscription_receiver.session
+        msgs = subscription_receiver.receive(max_batch_size=10)
+        session.set_session_state("BEGIN")
+        for msg in msgs:
+            msg.complete()
+            session.renew_lock()
+        session.set_session_state("END")
+
+### Rule operation
+
+# 1. get rules
+subscription_rule_manager = sb_client.get_subscription_rule_manager("topic_name", "subscription_name")
+rules = subscription_rule_manager.get_rules()
+for rule in rules:
+    print(rule.name)
+    print(rule.filter)  # If it's type of correlation, print the detail info
+    print(rule.action)
+
+# 2. add rule
+
+# 2.1 add boolean filter rule
+
+subscription_rule_manager.add_rule("rule_name", filter=True, sql_rule_action_expression=None)
+
+# 2.2 add sql filter rule
+
+message = Message("msg")
+message.correlation_id = 2
+message.user_properties = {"priority" : 2}
+subscription_rule_manager.add_rule("rule_name", filter="priority >= 1 AND sys.correlation-id = 2", sql_rule_action_expression=None)
+
+# 2.3 add correlation filter rule
+
+message = Message("msg")
+message.correlation_id = 2
+message.user_properties = {"key" : "value"}
+
+correlation_filter = CorrelationFilter()
+correlation_filter.user_properties = {"key": "value"}
```

Comment:
correlation_filter being read-only dict
need to be more **self-explain**
option: add_user_property_filer(key, value)